builtin/providers/azure/resource_azure_dns_server.go

```diff
@@ -0,0 +1,243 @@
+		})
+
+	// send the configuration back to Azure:
+	log.Println("[INFO] Sending updated network configuration back to Azure.")
+	reqID, err := networkClient.SetVirtualNetworkConfiguration(netConf)
+	if err != nil {
+		return fmt.Errorf("Failed issuing update to network configuration: %s", err)
+	}
+	err = managementClient.WaitForOperation(reqID, nil)
+	if err != nil {
+		return fmt.Errorf("Error setting network configuration: %s", err)
+	}
+
+	d.SetId(getRandomStringLabel(50))
+	return nil
+}
+
+// resourceAzureDnsServerRead does all the necessary API calls to read
+// the state of the DNS server off Azure.
+func resourceAzureDnsServerRead(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	log.Println("[INFO] Fetching current network configuration from Azure.")
+	netConf, err := networkClient.GetVirtualNetworkConfiguration()
+	if err != nil {
+		return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+	}
+
+	var found bool
+	name := d.Get("name").(string)
+
+	// search for our DNS and update it if the IP has been changed:
+	for _, dns := range netConf.Configuration.DNS.DNSServers {
+		if dns.Name == name {
+			found = true
+			d.Set("dns_address", dns.IPAddress)
+		}
+	}
+
+	// remove the resource from the state if it has been deleted in the meantime:
+	if !found {
+		d.SetId("")
+	}
+
+	return nil
+}
+
+// resourceAzureDnsServerUpdate does all the necessary API calls
+// to update the DNS definition on Azure.
+func resourceAzureDnsServerUpdate(d *schema.ResourceData, meta interface{}) error {
+	azureClient, ok := meta.(*AzureClient)
+	if !ok {
+		return fmt.Errorf("Failed to convert to *AzureClient, got: %T", meta)
+	}
+	managementClient := azureClient.managementClient
+	networkClient := virtualnetwork.NewClient(managementClient)
+
+	var err error
+	var found bool
+	name := d.Get("name").(string)
+	caddress := d.HasChange("dns_address")
+	var netConf virtualnetwork.NetworkConfiguration
+
+	if caddress {
+		log.Println("[DEBUG] DNS server address has changes; updating it on Azure.")
+		log.Println("[INFO] Fetching current network configuration from Azure.")
+		azureClient.mutex.Lock()
+		defer azureClient.mutex.Unlock()
+		netConf, err = networkClient.GetVirtualNetworkConfiguration()
+		if err != nil {
+			return fmt.Errorf("Failed to get the current network configuration from Azure: %s", err)
+		}
+
+		// search for our DNS and update its address value:
+		for i, dns := range netConf.Configuration.DNS.DNSServers {
+			found = true
+			if dns.Name == name {
+				netConf.Configuration.DNS.DNSServers[i].IPAddress = d.Get("dns_address").(string)
+			}
+		}
+
+		// if the config has changes, send the configuration back to Azure:
+		if found && caddress {
```

Comment:
You already did check if `caddress` is true on line 132, so doing it again here makes no sense...